api/service.go

```diff
@@ -893,91 +930,60 @@ func (s *service) CatFile(req *pb.CatFileRequest, server pb.API_CatFileServer) e
 	return nil
 }
 
-// RemoveFile handles a remove file request.
-func (s *service) RemoveFile(ctx context.Context, req *pb.RemoveFileRequest) (*pb.RemoveFileReply, error) {
-	log.Debugf("received remove file request")
+// RemoveBucketPath handles a remove bucket path request.
+func (s *service) RemoveBucketPath(
+	ctx context.Context,
+	req *pb.RemoveBucketPathRequest,
+) (*pb.RemoveBucketPathReply, error) {
+	log.Debugf("received remove bucket path request")
 
 	scope, ok := ctx.Value(reqKey("scope")).(string)
 	if !ok {
 		log.Fatal("scope required")
 	}
-	folderName, fileName := parsePath(req.Path)
-	folder, err := s.getFolderWithScope(ctx, folderName, scope)
+	buckName, fileName := parsePath(req.Path)
+	buck, err := s.getBucketWithScope(ctx, buckName, scope)
 	if err != nil {
 		return nil, err
 	}
-	folderPath := path.New(folder.Path)
 
-	dirpth, err := s.ipfsClient.Object().RmLink(ctx, folderPath, fileName)
+	buckPath := path.New(buck.Path)
+	dirpth, err := s.ipfsClient.Object().RmLink(ctx, buckPath, fileName)
```

Comment:
I wonder if we should check that someone isn't deleting `.seed`.
This check might also apply to pushing that file. Despite its main intention was to make buckets unique per project, if we don't error in that case we might get some unwanted consequence in a later _Remove_ operation. Since we consider that `.seed` is ephemeral and will be wiped-out if its the remaining file in the folder.

I guess `.seed` is fine, but maybe prefixing it with something extra may lower the chances that someone has some real `.seed` file and this gets in their way. e.g: `.textilebucketseed`?.

Reply:
Ah yeah, good catch. I'll add some additional checks for deletes and pushes, and change the name.